esptool.py

```diff
@@ -2433,15 +2443,30 @@ def add_spi_flash_subparsers(parent, is_elf2image):
 
     if operation_args[0] == 'esp':  # operation function takes an ESPLoader connection object
         initial_baud = min(ESPLoader.ESP_ROM_BAUD, args.baud)  # don't sync faster than the default baud rate
-        if args.chip == 'auto':
-            esp = ESPLoader.detect_chip(args.port, initial_baud, args.before, args.trace)
-        else:
-            chip_class = {
-                'esp8266': ESP8266ROM,
-                'esp32': ESP32ROM,
-            }[args.chip]
-            esp = chip_class(args.port, initial_baud, args.trace)
-            esp.connect(args.before)
+        ser_list = all_serial_ports() if args.port is None else [args.port]
+        ser_attempts = 0
+        for each_port in reversed(ser_list):
+            ser_attempts += 1
```

Comment:
This variable is no longer used for anything

Reply:
Sorted